lib/mshoplib/src/MShop/Plugin/Provider/Order/PropertyAdd.php

```diff
@@ -121,18 +121,19 @@ public function update( \Aimeos\MW\Observer\Publisher\Iface $order, string $acti
 		if( !is_array( $value ) )
 		{
 			\Aimeos\MW\Common\Base::checkClass( \Aimeos\MShop\Order\Item\Base\Product\Iface::class, $value );
-			return $this->addAttributes( $value, $this->getProductItems( [$value->getProductId()] ), $types );
+			return $this->addAttributes( $value, $this->getProductItems( [$value->getProductId()], [$value->getProductCode()]), $types );
 		}
 
 		$list = [];
 
 		foreach( $value as $orderProduct )
 		{
-			\Aimeos\MW\Common\Base::checkClass( \Aimeos\MShop\Order\Item\Base\Product\Iface::class, $orderProduct );
-			$list[] = $orderProduct->getProductId();
-		}
+			\Aimeos\MW\Common\Base::checkClass(\Aimeos\MShop\Order\Item\Base\Product\Iface::class, $orderProduct);
+			$list['code'][] = $orderProduct->getProductCode();
+			$list['id'][] = $orderProduct->getProductId();
+        	}
```

Comment:
You can shorten this to:
```
$ids = map( $value )->getProductId();
$codes = map( $value )->getProductCode();
```

Reply:
Do we still need to keep the checks for orderProduct(s)? ie. can do you want me to also remove the loop?

Reply:
Don't think they are really necessary

Reply:
> You can shorten this to:
> 
> ```
> $ids = map( $value )->getProductId();
> $codes = map( $value )->getProductCode();
> ```

This is sorcery...it's pretty cool to be able to call a method on an array of objects directly from the list.

Reply:
jQuery style ;-)